editor/project_manager.cpp

```diff
@@ -877,6 +878,42 @@ void ProjectManager::add_new_tag(const String &p_tag) {
 
 // Project converter/migration tool.
 
+void ProjectManager::_minor_project_migrate() {
+#ifndef DISABLE_DEPRECATED
+	Vector<ProjectList::Item> selected_list = project_list->get_selected_projects();
+	if (selected_list.size() != 1) {
+		// Don't perform migration when opening multiple projects.
+		_open_selected_projects();
+		return;
+	}
+
+	const ProjectList::Item &opened_project = selected_list[0];
+	if (version_convert_feature.begins_with("4.3")) {
+		// Migrate layout after scale changes.
+		const float edscale = EDSCALE;
+		if (edscale != 1.0) {
+			Ref<ConfigFile> layout_file;
+			layout_file.instantiate();
+
+			const String layout_path = opened_project.path.path_join(".godot/editor/editor_layout.cfg");
+			Error err = layout_file->load(layout_path);
+			if (err == OK) {
+				for (int i = 0; i < 4; i++) {
+					const String key = "dock_hsplit_" + itos(i + 1);
+					int old_value = layout_file->get_value("docks", key, 0);
+					if (old_value != 0) {
+						layout_file->set_value("docks", key, old_value / edscale);
+					}
+				}
+				layout_file->save(layout_path);
+			}
+		}
+	}
```

Comment:
I'd suggest moving that code to a `_minor_project_migrate` method, and keeping the name of the `pressed` callback more generic, like `_open_selected_projects_with_migration`.

This would be a better separation of concern IMO (the outcome should be exactly the same of course).